packages/assets-controllers/src/CurrencyRateController.ts

```diff
@@ -196,23 +197,27 @@ export class CurrencyRateController extends StaticIntervalPollingController<
           error.message.includes('market does not exist for this coin pair')
         )
       ) {
+        // Don't update state on transient / unexpected errors
+        shouldUpdateState = false;
         throw error;
       }
     } finally {
       try {
-        this.update(() => {
-          return {
-            currencyRates: {
-              ...currencyRates,
-              [nativeCurrency]: {
-                conversionDate,
-                conversionRate,
-                usdConversionRate,
+        if (shouldUpdateState) {
```

Comment:
Just wondering if it wouldn't be a more common pattern, if we just want to update the state when an error is not thrown, we can update the state as the last line of the try block, and have the finally just to release the mutex

Reply:
we want the state updated on some kids of errors though.  If we get `market does not exist for this coin pair`, then we want to update it to null.  Trying to maintain that existing behavior.

Reply:
Ah I see! Thanks for the explanation!